storage/copy_test.go

```diff
@@ -69,3 +70,72 @@ func TestCopyBothEncryptionKeys(t *testing.T) {
 		t.Errorf(`got %q, want it to contain "KMS"`, err)
 	}
 }
+
+// This test checks that request tokens are properly populated for Copy
+func TestCopyTokens(t *testing.T) {
+	ctx := context.Background()
+
+	expectedTokens := make(chan string, 10)
+	sendTokens := make(chan string, 10)
+
+	client := Client{
+		tc: mockRewriteObject{
+			expectedTokens: expectedTokens,
+			sendTokens:     sendTokens,
+		},
+	}
+
+	expectedTokens <- "" // first token should always be empty
+	sendTokens <- "atoken"
+	expectedTokens <- "atoken"
+	sendTokens <- "last123tokenexpected"
+	expectedTokens <- "last123tokenexpected"
+	sendTokens <- "sendToReqThatFinishes"
+	sendTokens <- "lastTokenToResume"
+
+	close(expectedTokens)
+	close(sendTokens)
+
+	dest := client.Bucket("b").Object("d").Key(testEncryptionKey)
+	c := dest.CopierFrom(client.Bucket("b").Object("s"))
+
+	if _, err := c.Run(ctx); err != nil {
+		t.Fatalf("err %v", err)
+	}
+
+	if got, want := c.RewriteToken, "lastTokenToResume"; got != want {
+		t.Errorf("mismatched token from copier; got: %s, want: %s", got, want)
+	}
+}
+
+type mockRewriteObject struct {
+	storageClient
+	expectedTokens chan string
+	sendTokens     chan string
+}
+
+func (m mockRewriteObject) RewriteObject(ctx context.Context, req *rewriteObjectRequest, opts ...storageOption) (*rewriteObjectResponse, error) {
```

Comment:
As discussed offline, I believe you can use maxBytesRewrittenPerCall from https://cloud.google.com/storage/docs/json_api/v1/objects/rewrite#parameters to force a multi-request copy in an integration test, rather than relying on mocks. See https://github.com/googleapis/google-api-go-client/blob/main/storage/v1/storage-gen.go#L10961